### PR TITLE
 tests/libnode: Unify exec command on node functions

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Continuing #13104 

The functions `ExecuteCommandInVirtHandlerPod` and `ExecuteCommandOnNodeThroughVirtHandler` are pretty much the same.
The only difference is `ExecuteCommandOnNodeThroughVirtHandler` returning the command stderr as is.
While `ExecuteCommandInVirtHandlerPod` combine the stderr as part of the returned error.

Since in all `ExecuteCommandOnNodeThroughVirtHandler` usages, the returned stderr is printed in one way or another (e.g.:  reported  it though ginkgoExpect message).
There is no value in returning it over having stderr part of the returned error message.

Hence, merge `ExecuteCommandOnNodeThroughVirtHandler` into `ExecuteCommandInVirtHandlerPod`.

Before this PR:
Some tests use `ExecuteCommandOnNodeThroughVirtHandler` and others `ExecuteCommandInVirtHandlerPod`.

After this PR:
All tests use `ExecuteCommandInVirtHandlerPod`.
`ExecuteCommandOnNodeThroughVirtHandler` is removed.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by ormergi** [comment]
@Barakmor1 following your comment on the original PR https://github.com/kubevirt/kubevirt/pull/13104#issuecomment-2545666962, please see second commit it mitigates the lake of stderr for debugging.
Please note there are places that ignored the stderr/out originally, the behavior in this places remain the same.

File: tests/infrastructure/virt-handler.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ var _ = DescribeSerialInfra("virt-handler", func() {
 		nodesWithKSM := make([]string, 0)
 		for _, node := range nodes.Items {
 			command := []string{"cat", "/sys/kernel/mm/ksm/run"}
-			_, err := libnode.ExecuteCommandInVirtHandlerPod(node.Name, command)
+			_, _, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(node.Name, command)
```
Comment (Barakmor1) [inline tests/infrastructure/virt-handler.go:59]: Hey, Thanks for the PR.
I think that we are still losing logs here

File: tests/launchsecurity/sev.go [hunk 2]
```diff
@@ -270,14 +270,14 @@ var _ = Describe("[sig-compute]AMD Secure Encrypted Virtualization (SEV)", decor
 			Expect(nodeName).ToNot(BeEmpty())
 
 			checkCmd := []string{"ls", sevDevicePath}
-			_, err = libnode.ExecuteCommandInVirtHandlerPod(nodeName, checkCmd)
+			_, _, err = libnode.ExecuteCommandOnNodeThroughVirtHandler(nodeName, checkCmd)
```
Comment (Barakmor1) [inline tests/launchsecurity/sev.go:273]: and here

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

File: tests/storage/scsidisk.go [hunk 3]
```diff
@@ -83,8 +83,8 @@ func CreateSCSIDisk(nodeName string, opts []string) (address string, device stri
 func RemoveSCSIDisk(nodeName, address string) {
 	By("Removing scsi disk")
 	args := []string{"/usr/bin/echo", "1", ">", fmt.Sprintf("/proc/1/root/sys/class/scsi_device/%s/device/delete", address)}
-	_, err := libnode.ExecuteCommandInVirtHandlerPod(nodeName, args)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to disable scsi disk")
+	stdout, stderr, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(nodeName, args)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to disable scsi disk, %s, %s", stdout, stderr))
```
Comment (Barakmor1) [inline tests/storage/scsidisk.go:87]:
```suggestion
	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to disable scsi disk, stdout:%s, stderr:%s", stdout, stderr))
```

File: tests/storage/scsidisk.go [hunk 4]
```diff
@@ -93,14 +93,14 @@ func RemoveSCSIDisk(nodeName, address string) {
 
 func FixErrorDevice(nodeName string) {
 	args := []string{"/usr/bin/bash", "-c", "echo 0 > /proc/1/root/sys/bus/pseudo/drivers/scsi_debug/opts"}
-	stdout, err := libnode.ExecuteCommandInVirtHandlerPod(nodeName, args)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fix faulty disk, %s", stdout))
+	stdout, stderr, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(nodeName, args)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fix faulty disk, %s, %s", stdout, stderr))
```
Comment (Barakmor1) [inline tests/storage/scsidisk.go:97]:
```suggestion
	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fix faulty disk, stdout:%s, stderr:%s", stdout, stderr))
```

File: tests/storage/scsidisk.go [hunk 5]
```diff
@@ -93,14 +93,14 @@ func RemoveSCSIDisk(nodeName, address string) {
...
 
 	args = []string{"/usr/bin/cat", "/proc/1/root/sys/bus/pseudo/drivers/scsi_debug/opts"}
 
 	By("Checking opts of scsi_debug")
-	stdout, err = libnode.ExecuteCommandInVirtHandlerPod(nodeName, args)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to fix faulty disk")
+	stdout, stderr, err = libnode.ExecuteCommandOnNodeThroughVirtHandler(nodeName, args)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fix faulty disk, %s, %s", stdout, stderr))
```
Comment (Barakmor1) [inline tests/storage/scsidisk.go:103]:
```suggestion
	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to fix faulty disk, stdout:%s, stderr:%s", stdout, stderr))
```

**Review by Barakmor1 — COMMENTED** [review]
Hey @ormergi,
Thanks for the PR! I still think it would be better to use `ExecuteCommandInVirtHandlerPod` and deprecate `ExecuteCommandOnNodeThroughVirtHandler` to avoid all the boilerplate around logging in the tests.

**Comment by ormergi** [comment]
> Hey @ormergi, Thanks for the PR! I still think it would be better to use `ExecuteCommandInVirtHandlerPod` and deprecate `ExecuteCommandOnNodeThroughVirtHandler` to avoid all the boilerplate around logging in the tests.

Looking at this again I agree `ExecuteCommandInVirtHandlerPod` is easier to work with and it"ll save handling the stderr.
I will do that.

**Comment by ormergi** [comment]
> > Hey @ormergi, Thanks for the PR! I still think it would be better to use `ExecuteCommandInVirtHandlerPod` and deprecate `ExecuteCommandOnNodeThroughVirtHandler` to avoid all the boilerplate around logging in the tests.
> 
> Looking at this again I agree `ExecuteCommandInVirtHandlerPod` is easier to work with and it"ll save handling the stderr. I will do that.

Done
@Barakmor1 / @ShellyKa13 PTAL

**Comment by Barakmor1** [comment]
Thanks @ormergi , it looks much better now!
/lgtm

**Comment by akalenyu** [comment]
/test pull-kubevirt-e2e-k8s-1.30-sig-storage

@ShellyKa13 This is a known flake for us; what happens is that this field only gets populated a while after the VM is created, so the test runs into a webhook failure when posting the VMClone.
Do you think we should tackle this before the webhook disappears?

**Comment by ormergi** [comment]
I accidentally closed the PR, it has been reopened. No changes has been done though.
Sorry for the noise.

/retest-required

**Comment by ormergi** [comment]
[SR-IOV lane failure](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/13590/pull-kubevirt-e2e-kind-sriov/1871311358797025280) is know issue tracked by https://github.com/kubevirt/kubevirt/issues/12691

/test pull-kubevirt-e2e-kind-sriov

**Comment by ShellyKa13** [comment]
@akalenyu @iholder101 Regarding the test:
`VirtualMachineClone Tests VM clone [sig-storage]with more complicated VM and snapshot storage class double cloning: clone target as a clone source`
As @akalenyu commented https://github.com/kubevirt/kubevirt/pull/13590#issuecomment-2560091360
I have a plan to remove the webhook all together, which should be soon, I think we can quarantine this test for now if this is flaky and I will unquarantine it in the PR that removes the webhook. WDYT?
The fix for the test should be simple but not sure if its worth fixing it just to remove the fix in the near future

**Comment by iholder101** [comment]
> @akalenyu @iholder101 Regarding the test: `VirtualMachineClone Tests VM clone [sig-storage]with more complicated VM and snapshot storage class double cloning: clone target as a clone source` As @akalenyu commented [#13590 (comment)](https://github.com/kubevirt/kubevirt/pull/13590#issuecomment-2560091360) I have a plan to remove the webhook all together, which should be soon, I think we can quarantine this test for now if this is flaky and I will unquarantine it in the PR that removes the webhook. WDYT? The fix for the test should be simple but not sure if its worth fixing it just to remove the fix in the near future

Your call, sounds good to me.
Just please create a tracking issue for that if you don't mind

**Comment by akalenyu** [comment]
> @akalenyu @iholder101 Regarding the test: `VirtualMachineClone Tests VM clone [sig-storage]with more complicated VM and snapshot storage class double cloning: clone target as a clone source` As @akalenyu commented [#13590 (comment)](https://github.com/kubevirt/kubevirt/pull/13590#issuecomment-2560091360) I have a plan to remove the webhook all together, which should be soon, I think we can quarantine this test for now if this is flaky and I will unquarantine it in the PR that removes the webhook. WDYT? The fix for the test should be simple but not sure if its worth fixing it just to remove the fix in the near future

I think it will make things better but I am not entirely sure about this test being the only occurrence of this flake
but yeah up to you

**Pull request merged**